packages/liveblocks-node/src/thread-notification-helpers.ts

```diff
@@ -0,0 +1,284 @@
+import type {
+  BaseUserMeta,
+  CommentBody,
+  CommentBodyJson,
+  CommentData,
+  DRI,
+  DU,
+  OptionalPromise,
+  ResolveUsersArgs,
+} from "@liveblocks/core";
+import {
+  generateCommentUrl,
+  getMentionedIdsFromCommentBody,
+  transformCommentBody,
+} from "@liveblocks/core";
+
+import type { Liveblocks } from "./client";
+import type { ThreadNotificationEvent } from "./webhooks";
+
+export type ThreadNotificationCommentData = Omit<
+  CommentData,
+  "body" | "deletedAt"
+> & {
+  body: CommentBody;
+  deletedAt?: never;
+};
+
+export type ThreadNotificationUnreadRepliesData = {
+  type: "unreadReplies";
+  comments: ThreadNotificationCommentData[];
+};
+
+export type ThreadNotificationUnreadMentionData = {
+  type: "unreadMention";
+  comments: ThreadNotificationCommentData[];
+};
+
+export type ThreadNotificationData =
+  | ThreadNotificationUnreadRepliesData
+  | ThreadNotificationUnreadMentionData;
+
+/** @internal */
+export const getLastCommentWithMention = ({
+  comments,
+  mentionedUserId,
+}: {
+  comments: ThreadNotificationCommentData[];
+  mentionedUserId: string;
+}): ThreadNotificationCommentData | null => {
+  return (
+    Array.from(comments)
+      .reverse()
+      .filter((c) => c.userId !== mentionedUserId)
+      .find((c) => {
+        const mentionedUserIds = getMentionedIdsFromCommentBody(c.body);
+        return mentionedUserIds.includes(mentionedUserId);
+      }) ?? null
+  );
+};
+
+/**
+ * Get thread notification data helper.
+ *
+ * It returns either an object containing a list of unread replies for a thread
+ * or either an object containing a list of the last unread comment where the notification
+ * receiver was mentioned in.
+ *
+ * @param params.client Liveblocks node client
+ * @param params.event The thread notification event
+ * @returns A thread notification data object
+ *
+ * @example Unread replies:
+ * {
+ *  type: "unreadReplies",
+ *  comments: [unread_comment_0, unread_comment_1, ...],
+ * }
+ *
+ * @example Unread mention:
+ * {
+ *  type: "unreadMention",
+ *  comments: [unread_comment_with_mention]
+ * }
+ */
+export async function getThreadNotificationData(params: {
+  client: Liveblocks;
+  event: ThreadNotificationEvent;
+}): Promise<ThreadNotificationData> {
```

Comment:
I don't think there's a clear answer, it's only opinions for this and mine is that if the function would only do some data formatting, it should be a separate helper, but if it does call other methods within, I would be fine with having it on the client. And the docs would not have this paragraph to make it clear that it's not a REST API.

![api](https://github.com/user-attachments/assets/8a1d4032-e41f-4117-89c5-40e8063fee2a)

Reply:
Interesting choice of API design 🤔 
I was checking the code of `broadcastEvent` and I see it does well a `this.post` and only throws in case of errors. No response at all 😮 

I well understand the what here but I do not clearly understand the why here.
From my perspective it's not really a good API design here 🤔 I mean I'm okay we just broadcast an event and we only need to know when something is wrong.

But in the dashboard I implementation the internal route and I return well a status when is everything is good due to UI/UX requirements and needs. 
How our customers do the same to fulfill UI/UX requirements if they rely on `@liveblocks/node? 😮